### PR TITLE
Support custom rule classes in `@rules` and `@rulesForArray`

### PR DESCRIPTION
**Pull request by omarusman**
- [x] Added or updated tests
- [x] Added Docs for all relevant versions
- [x] Updated the changelog

Support or allow custom rules in @rules directive.

**Usage**
```cli
php artisan make:rule MyCustomRule
```
and in your _schema.graphql_
```graphql
input FooBar {
    field: String! @rules(apply: ["required", "App\\Rules\\MyCustomRule"])
}
```

### CODE REVIEW

File: src/Schema/Directives/RulesDirective.php [hunk 1]
```diff
@@ -28,6 +28,13 @@ public function getRules(): array
     {
         $rules = $this->directiveArgValue('apply');
 
+        // Resolve custom rule namespace, if possible.
+        foreach ($rules as $key => $rule) {
+            if (strpos($rule, 'App\Rules') === 0) {
```
Comment (spawnia) [inline src/Schema/Directives/RulesDirective.php]: I think we might just use `class_exists()` to determine if we are dealing with a custom rule.

**Comment by spawnia** [comment]
@omarusman hey, thanks for the PR. I like the idea behind this, the approach is elegant.

In order for this to be merged, make sure you can check the boxes in the PR template (i added them back in to the description.). Tests, Docs and the Changelog should be in place.

In addition to the `@rules` directive, the `@rulesForArray` directive should also receive an update.

**Comment by omarusman** [comment]
@spawnia Made some changes. Let me know what do you think. 😊

**Comment by chrissm79** [comment]
This is awesome, thanks @omarusman and @spawnia!

**Comment by omarusman** [comment]
@spawnia  cool 😊

**Pull request merged**